docs/design/schema-compatibility.md

```diff
@@ -0,0 +1,119 @@
+# Support for forward and backward compatibility in DCL schemes
+
+Schema changes can cover a wide range of modifications with varying impacts on application compatibility and data integrity. Below are use cases with strategies to manage schema changes and ensure compatibility.
+
+## Multiple versions can live in parallel
```

Comment:
It would be great to add number for every option.
Like

I. Multiple versions can live in parallel
  1. Strategy for Compatible Changes
     - Option A
     - Option B
  2.  Strategy for Non-Compatible Changes
       - Option C
       - Option D

II. New version replaces the legacy one (V2 replaces V1)
   1. Strategy for Compatible or Convertable changes
        - Option E: Not keeping backward compatibility in API
        - Option F: Keeping backward compatibility in API
   2. Strategy for Non-convertable changes
        - Option G: Not keeping backward compatibility in API
        - Option H: Keeping backward compatibility in API